tests/Utils/SchemaExtenderTest.php

```diff
@@ -0,0 +1,1929 @@
+                DirectiveLocation::OBJECT,
+                DirectiveLocation::FIELD_DEFINITION,
+                DirectiveLocation::ARGUMENT_DEFINITION,
+                DirectiveLocation::IFACE,
+                DirectiveLocation::UNION,
+                DirectiveLocation::ENUM,
+                DirectiveLocation::ENUM_VALUE,
+                DirectiveLocation::INPUT_OBJECT,
+                DirectiveLocation::INPUT_FIELD_DEFINITION,
+            ],
+        ]);
+
+        $this->testSchema = new Schema([
+            'query' => new ObjectType([
+                'name' => 'Query',
+                'fields' => static function () use ($FooType, $SomeScalarType, $SomeUnionType, $SomeEnumType, $SomeInterfaceType, $SomeInputType) {
+                    return [
+                        'foo' => [ 'type' => $FooType ],
+                        'someScalar' => [ 'type' => $SomeScalarType ],
+                        'someUnion' => [ 'type' => $SomeUnionType ],
+                        'someEnum' => [ 'type' => $SomeEnumType ],
+                        'someInterface' => [
+                            'args' => [
+                                'id' => [
+                                    'type' => Type::nonNull(Type::ID()),
+                                ],
+                            ],
+                            'type' => $SomeInterfaceType,
+                        ],
+                        'someInput' => [
+                            'args' => [ 'input' => [ 'type' => $SomeInputType ] ],
+                            'type' => Type::string(),
+                        ],
+                    ];
+                },
+            ]),
+            'types' => [$FooType, $BarType],
+            'directives' => array_merge(GraphQL::getStandardDirectives(), [$FooDirective]),
+        ]);
+
+        $testSchemaAst = Parser::parse(SchemaPrinter::doPrint($this->testSchema));
+
+        $this->testSchemaDefinitions = array_map(static function ($node) {
+            return Printer::doPrint($node);
+        }, iterator_to_array($testSchemaAst->definitions->getIterator()));
+
+        $this->FooDirective = $FooDirective;
+        $this->FooType      = $FooType;
+    }
+
+    protected function dedent(string $str) : string
+    {
+        $trimmedStr = trim($str, "\n");
+        $trimmedStr = preg_replace('/[ \t]*$/', '', $trimmedStr);
+
+        preg_match('/^[ \t]*/', $trimmedStr, $indentMatch);
+        $indent = $indentMatch[0];
+        return preg_replace('/^' . $indent . '/m', '', $trimmedStr);
+    }
+
+    /**
+     * @param mixed[]|null $options
+     */
+    protected function extendTestSchema(string $sdl, ?array $options = null) : Schema
+    {
+        $originalPrint  = SchemaPrinter::doPrint($this->testSchema);
+        $ast            = Parser::parse($sdl);
+        $extendedSchema = SchemaExtender::extend($this->testSchema, $ast, $options);
+        $this->assertEquals(SchemaPrinter::doPrint($this->testSchema), $originalPrint);
```

Comment:
can assertions be called statically where appropriate?

For ref see https://github.com/webonyx/graphql-php/pull/354#issuecomment-422900157